### PR TITLE
added Delphi variant

### PR DESCRIPTION
**Pull request by ansgargosling**
Hi there,

I added a Version with delphi-sourcecode,

### CODE REVIEW

**Comment by ansgargosling** [comment]
Added Testcases
Fixed Currency-Output

**Comment by codecop** [comment]
Thank you. Nice to see some Pascal. The code looks good. I added some review comments to the code, please address them.

**Comment by ansgargosling** [comment]
> Thank you. Nice to see some Pascal. The code looks good. I added some review comments to the code, please address them.

Hi codecop,

thanks for your feedback. Maybe a dumb question, but where can i find your comments? This is my first contribution to an existing project, I would appreciate your help

**Review by codecop — CHANGES_REQUESTED** [review]
I do not know how to use GitHub review feature...

**Comment by codecop** [comment]
Sorry I had not submitted the "review" but I was seeing it so I thought you would be seeing it too. Now it is here

File: delphi/main/TheatricalPlayers.dproj
[the remark this answers is not in the document]
Comment (ansgargosling) [reply delphi/main/TheatricalPlayers.dproj:1]: Yes, this is a clean new project. I know, the dproj-files are a bit messy.

File: delphi/main/clrLoadData.pas [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+type
+  TLoadData = class(TObject)
+    class procedure LoadPLays(APLays: TPlays);
+    class procedure LoadInvoices(AInvoices: TInvoices);
+  end;
+
+implementation
+
+
```
[the remark this answers is not in the document]
Comment (ansgargosling) [reply delphi/main/clrLoadData.pas:21]: Done!

File: delphi/main/clrStatementPrinter.pas [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    // print line for this order
+    Result := Result + Format('  %s: %s (%d seats)'#13#10, [Play.Name, FormatCurrency(ThisAmount / 100), Performance.Audience]);
+    TotalAmount := TotalAmount + ThisAmount;
+  end;
+
+  Result := Result + Format('Amount owed is %s'#13#10, [FormatCurrency(TotalAmount / 100)]);
+  Result := Result + Format('You earned %d credits'#13#10, [VolumeCredits]);
+end;
+
+
```
[the remark this answers is not in the document]
Comment (ansgargosling) [reply delphi/main/clrStatementPrinter.pas:84]: Done

File: delphi/test/Win32/Debug/dunitx-results.xml
[the remark this answers is not in the document]
Comment (ansgargosling) [reply delphi/test/Win32/Debug/dunitx-results.xml:1]: Added to ignore-list

File: delphi/test/uTestStatementPrinter.pas
[the remark this answers is not in the document]
Comment (ansgargosling) [reply delphi/test/uTestStatementPrinter.pas:1]: Done!

File: delphi/test/uTestStatementPrinter.pas [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+  Plays: TPlays;
+  Performances: TPerformances;
+const
+  Expected = '''
+  Statement for BigCo
+    Hamlet: $650.00 (55 seats)
+    As You Like It: $580.00 (35 seats)
+    Othello: $500.00 (40 seats)
+  Amount owed is $1,730.00
+  You earned 47 credits
```
[the remark this answers is not in the document]
Comment (ansgargosling) [reply delphi/test/uTestStatementPrinter.pas:125]: Hmm? Didn't see this acually in my code. Try to push it again

**Comment by ansgargosling** [comment]
no problem, you're a human, not a robot ;)
the changes are commited

**Comment by codecop** [comment]
This looks good. Vote to merge @emilybache

**Comment by emilybache** [comment]
Thankyou both for all your hard work.

**Pull request merged**